nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -359,131 +385,179 @@ in
-        '';
-      };
+        systemd.services.slurmdbd =
+          let
+            # slurm strips the last component off the path
+            configPath = "$RUNTIME_DIRECTORY/slurmdbd.conf";
+          in
+          lib.mkIf (cfg.dbdserver.enable) {
+            path = with pkgs; [
+              wrappedSlurm
+              munge
+              coreutils
+            ];
+
+            wantedBy = [ "multi-user.target" ];
+            after = [
+              "network.target"
+              "munged.service"
+              "mysql.service"
+            ];
+            requires = [
+              "munged.service"
+              "mysql.service"
+            ];
+
+            preStart = ''
+              install -m 600 -o ${cfg.user} -T ${slurmdbdConf} ${configPath}
+              ${lib.optionalString (cfg.dbdserver.storagePassFile != null) ''
+                echo "StoragePass=$(cat ${cfg.dbdserver.storagePassFile})" \
+                  >> ${configPath}
+              ''}
+            '';
+
+            script = ''
+              export SLURM_CONF=${configPath}
+              exec ${cfg.package}/bin/slurmdbd -D
+            '';
+
+            serviceConfig = {
+              RuntimeDirectory = "slurmdbd";
+              Type = "simple";
+              PIDFile = "/run/slurmdbd.pid";
+              ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+            };
+          };
 
-      systemd.services.slurmdbd =
-        let
-          # slurm strips the last component off the path
-          configPath = "$RUNTIME_DIRECTORY/slurmdbd.conf";
-        in
-        lib.mkIf (cfg.dbdserver.enable) {
+        systemd.services.slurmrestd = lib.mkIf (cfg.rest.enable) {
           path = with pkgs; [
             wrappedSlurm
-            munge
             coreutils
           ];
 
           wantedBy = [ "multi-user.target" ];
           after = [
-            "network.target"
-            "munged.service"
-            "mysql.service"
+            "systemd-tmpfiles-clean.service"
+            "network-online.target"
+            "remote-fs.target"
+            "slurmctld.service"
           ];
-          requires = [
-            "munged.service"
-            "mysql.service"
-          ];
-
-          preStart = ''
-            install -m 600 -o ${cfg.user} -T ${slurmdbdConf} ${configPath}
-            ${lib.optionalString (cfg.dbdserver.storagePassFile != null) ''
-              echo "StoragePass=$(cat ${cfg.dbdserver.storagePassFile})" \
-                >> ${configPath}
-            ''}
-          '';
-
-          script = ''
-            export SLURM_CONF=${configPath}
-            exec ${cfg.package}/bin/slurmdbd -D
-          '';
+          wants = [ "network-online.target" ];
 
           serviceConfig = {
-            RuntimeDirectory = "slurmdbd";
             Type = "simple";
-            PIDFile = "/run/slurmdbd.pid";
+            ExecStart = "${wrappedSlurm}/bin/slurmrestd $SLURMRESTD_OPTIONS";
```

Comment:
Where does `$SLURMRESTD_OPTIONS` come from (I can't find where it is set)?

Reply:
You are right that everything can be set with via the `extraEnv`, but that is not transparent to the user. I think a separate option would be better here.

Reply:
You're right, I'll update it to have a dedicated option.